src/deepdisc/model/loaders.py

```diff
@@ -86,69 +55,43 @@ def __call__(self, dataset_dict):
         }
 
 
-class test_mapper_cls:
-    def __init__(self, **read_image_args):
-        self.ria = read_image_args
+class redshift_train_mapper_cls:
+    def __init__(self, imreader, key_mapper):
+        self.IR = imreader
+        self.km = key_mapper
 
     def __call__(self, dataset_dict):
-        dataset_dict = copy.deepcopy(dataset_dict)  # it will be modified by code below
-        filenames = [
-            dataset_dict["filename_G"],
-            dataset_dict["filename_R"],
-            dataset_dict["filename_I"],
-        ]
+        dataset_dict = copy.deepcopy(dataset_dict)
+        key = self.km(dataset_dict)
 
-        if self.ria["normalize"] != "random":
-            image = toolkit.read_image_hsc(
-                filenames,
-                normalize=self.ria["normalize"],
-                ceil_percentile=self.ria["ceil_percentile"],
-                dtype=self.ria["dtype"],
-                A=self.ria["A"],
-                stretch=self.ria["stretch"],
-                Q=self.ria["Q"],
-                do_norm=self.ria["do_norm"],
-            )
-        elif self.ria["normalize"] == "random":
-            scalings = np.array(["astrolupton", "astroluptonhc", "zscore"])
-            norm = np.random.choice(scalings, 1)[0]
-            if norm == "zscore":
-                dtype = np.int16
-            else:
-                dtype = np.uint8
-            image = toolkit.read_image_hsc(
-                filenames,
-                normalize=norm,
-                ceil_percentile=self.ria["ceil_percentile"],
-                dtype=dtype,
-                A=self.ria["A"],
-                stretch=self.ria["stretch"],
-                Q=self.ria["Q"],
-                do_norm=self.ria["do_norm"],
-            )
+        image = self.IR(key)
 
-        augs = T.AugmentationList(
+        augs = detectron_addons.KRandomAugmentationList(
             [
-                T.CropTransform(
-                    image.shape[1] // 4,
-                    image.shape[0] // 4,
-                    image.shape[1] // 2,
-                    image.shape[0] // 2,
-                )
```

Comment:
To confirm, it looks like this PR moves the training from using a deterministic cropping to a random one. I think this makes sense, but wanted to check that it was intentional.

Reply:
Yeah, the change should be ok